web3.js/src/connection.ts

```diff
@@ -2945,9 +2959,46 @@ export class Connection {
           () => resolve({__type: TransactionStatus.TIMED_OUT, timeoutMs}),
           timeoutMs,
         );
+      } else if (Object.prototype.hasOwnProperty.call(strategy, 'nonceHash')) {
+        const {nonceAccount, nonceHash} =
+          strategy as NonceBasedTransactionConfirmationStrategy;
+        let currentNonceHash = nonceHash;
+        const getCurrentNonceHash = async () => {
+          try {
+            const nonceAccountData = await this.getNonce(
```

Comment:
This could return stale nonce account state. If an old nonce value is returned, the confirmation logic will assume that the mismatch is due to the nonce being advanced when it actually hasn't yet. To avoid stale state, the user can pass a `minContextSlot` for the nonce state. When constructing a durable nonce tx, users should use `getNonceAndContext` and pass the returned `context.slot` as `minContextSlot` during confirmation.